app/src/ui/history/commit-summary.tsx

```diff
@@ -15,14 +15,95 @@ interface ICommitSummaryProps {
   readonly files: ReadonlyArray<FileChange>
   readonly emoji: Map<string, string>
   readonly gitHubUser: IGitHubUser | null
+
+  /**
+   * Whether or not the commit body container should
+   * be rendered expanded or not. In expanded mode the
+   * commit body container takes over the diff view
+   * allowing for full height, scrollable reading of
+   * the commit message body.
+   */
   readonly isExpanded: boolean
+
   readonly onExpandChanged: (isExpanded: boolean) => void
 }
 
 interface ICommitSummaryState {
+  /**
+   * The commit message summary, i.e. the first line
+   * in the commit message. Note that this
+   * may differ from the body property in the commit object
+   * passed through props, see the createState method for more
+   * details.
+   */
+  readonly summary: string
+
+  /**
+   * The commit message body, i.e. anything after the
+   * first line of text in the commit message. Note that this
+   * may differ from the body property in the commit object
+   * passed through props, see the createState method for more
+   * details.
+   */
+  readonly body: string
+
+  /**
+   * Whether or not the commit body text overflows
+   * its container. Used in conjunction with the isExpanded
+   * prop
+   */
   readonly isOverflowed: boolean
 }
 
+const maxSummaryLength = 72
+
+/**
+ * Removes whitespace characters from the end of the
+ * string
+ */
+function trimTrailingWhitespace(value: string) {
+  return value.replace(/\s+$/, '')
+}
+
+/**
+ * Creates the state object for the CommitSummary component.
+ *
+ * Ensures that the commit summary never exceeds 72 characters
+ * and wraps it into the commit body if it does.
+ *
+ * @param isOverflowed Whether or not the component should render
+ *                     the commit body in expanded mode, see the
+ *                     documentation for the isOverflowed state
+ *                     property
+ *
+ * @param props        The current commit summary prop object.
+ */
+function createState(isOverflowed: boolean, props: ICommitSummaryProps) {
+  let summary = trimTrailingWhitespace(props.commit.summary)
+  let body = trimTrailingWhitespace(props.commit.body)
+
+  if (summary.length > maxSummaryLength) {
+    // Truncate at least 3 characters off the end to avoid just an ellipsis
+    // followed by 1-2 characters in the body. This matches dotcom behavior.
+    const truncateLength = maxSummaryLength - 3
```

Comment:
I know this matched dotcom behavior, but could we be greedy here and take the whole word?

e.g., 

```
Some text is too l...
ong.
```
Would instead be...
```
Some text is too...
long.
```

Reply:
I thought about doing something like that but I feel like we should start out simple. If we try for something greedy we run the risk of edge cases with long words and such. Following the dotcom behavior seems safe

Reply:
Mind pulling that `3` out into a named const?